### PR TITLE
Move Task Box method to Core & remove TaskUtility

### PR DESCRIPTION
**Pull request by ReubenBond**
Part of #3557

In generated invokers, we will use `return await <method call>` instead of `return <method call>.Box()`.

### CODE REVIEW

File: src/Orleans.Core/Async/TaskExtensions.cs [hunk 1]
```diff
@@ -1,13 +1,135 @@
...
+            return default(object);
+        }
+
+        private static async Task<object> BoxAwait<T>(Task<T> task)
+        {
+            return await task;
+        }
+
+        private static Task<T> UnboxContinuation<T>(Task<object> task)
+        {
```
Comment (ReubenBond) [inline src/Orleans.Core/Async/TaskExtensions.cs]: I believe there's room for improvement here and in related methods, but we can do that in another PR.

**Comment by ReubenBond** [comment]
The failures here are fixed in #3560

**Comment by jdom** [comment]
@dotnet-bot test functional

File: src/Orleans.Core/Async/TaskExtensions.cs (hunk 1 above)
Comment (jdom) [reply src/Orleans.Core/Async/TaskExtensions.cs]: Absolutely, we can also update codegen to avoid so much boxing, at least separating the paths for Task and TaskOfT
